src/components/sections/action-item.tsx

```diff
@@ -55,40 +63,76 @@ export default function ActionItem({
   return (
     <TouchableOpacity
       onPress={() => onPress?.(!checked)}
-      style={[style.spaceBetween, topContainer]}
+      style={[
+        style.spaceBetween,
+        topContainer,
+        {
+          backgroundColor: interactiveColor
+            ? interactiveColor.background
+            : topContainer.backgroundColor,
+        },
+      ]}
       testID={testID}
       accessibilityRole={role}
       accessibilityState={{
         [stateName]: checked,
       }}
       {...accessibility}
     >
-      <ThemeText
-        type={
-          mode === 'heading-expand' ? 'body__primary--bold' : 'body__primary'
-        }
-        style={contentContainer}
-      >
-        {text}
-      </ThemeText>
-      <ActionModeIcon mode={mode} checked={checked} />
+      <View style={{flexShrink: 1}}>
+        <ThemeText
+          type={
+            mode === 'heading-expand' ? 'body__primary--bold' : 'body__primary'
+          }
+          style={[
+            contentContainer,
+            interactiveColor ? {color: interactiveColor.text} : undefined,
+          ]}
+        >
+          {text}
+        </ThemeText>
+        {subtext && (
+          <ThemeText
+            type="body__secondary"
+            color="secondary"
+            style={{marginTop: theme.spacings.medium}}
+          >
+            {subtext}
+          </ThemeText>
+        )}
+      </View>
+      <ActionModeIcon
+        mode={mode}
+        checked={checked}
+        color={interactiveColor ? color : undefined}
+      />
     </TouchableOpacity>
   );
 }
 
 function ActionModeIcon({
   mode,
   checked,
-}: Pick<ActionItemProps, 'mode' | 'checked'>) {
+  color,
+}: Pick<ActionItemProps, 'mode' | 'checked' | 'color'>) {
   const style = useHeaderExpandStyle();
   const {t} = useTranslation();
+  const {theme} = useTheme();
 
   switch (mode) {
     case 'check': {
       if (checked) {
-        return <ThemeIcon svg={Confirm} />;
+        return (
+          <ThemeIcon
+            svg={Confirm}
+            {...(color
+              ? {fill: theme.interactive[color].active.text}
+              : undefined)}
+          />
+        );
       } else {
-        return null;
+        // In order to keep margins consistent, a transparent icon is returned.
+        return <ThemeIcon svg={Confirm} fill="#FFFFFF00" />;
```

Comment:
This is definitely hacky, but maybe an ok solution compared to calculating the margin in other ways. I'm of course open to other suggestions.

Reply:
I think it's ok. Maybe a little cleaner to add a parameter on `ThemeIcon`? Then you can skip the `if` and do something like this:
```typescript
<ThemeIcon
  ...
  transparent={!checked}
/>
```

Reply:
Agreed. Realized there already is a `fillOpacity` prop, so I used that instead of adding `transparent`.